### ISSUE TITLE
(native automation mode) .js file mocks transform file content to html code

### ISSUE THREAD

**Issue by Thomas-Haider** [DevExpress/testcafe#7823]
### What is your Scenario?

I want to prevent a js file from execution and therefore mock the request.

### What is the Current behavior?

When I use a request mock to mock a javascript file the content of the file gets transformed to html code. This leads to `SyntaxError: Unexpected token '<'` as it obviously can't handle html tags inside the js-file. Note: this only happens when the native automation mode is active.

Simple js file mock with response body set to {}:
```typescript
export const jsFileMock = RequestMock()
    .onRequestTo(/.*ruxitagentjs_.*\.js/g)
    .respond({}, 200);
```
results in the following js file content:
``` html
<html><head><link rel="stylesheet" type="text/css" class="ui-stylesheet-hammerhead-shadow-ui" href="http://localhost:61180/testcafe-ui-styles.css"><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://localhost:61180/hammerhead.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://localhost:61180/testcafe-core.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://localhost:61180/testcafe-ui.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://localhost:61180/testcafe-automation.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8" src="http://localhost:61180/testcafe-driver.js"></script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8">(function() {})()</script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8">Object.defineProperty(window, '%nativeAutomationContextStorage%', { configurable: true, value: "" });</script><script type="text/javascript" class="script-hammerhead-shadow-ui" charset="UTF-8">window['%hammerhead%'].utils.removeInjectedScript();

window['%hammerhead%'].start({
    cookie                   : "",
    sessionId                : 'YZdrO4azk',
    serviceMsgUrl            : 'http://localhost:61180/messaging',
    transportWorkerUrl       : 'http://localhost:61180/transport-worker.js',
    isFirstPageLoad          : false,
    crossDomainProxyPort     : '61181',
    referer                  : "",
    forceProxySrcForImage    : false,
    iframeTaskScriptTemplate : "window['%hammerhead%'].utils.removeInjectedScript();\n\nwindow['%hammerhead%'].start({\n    cookie                   : \"\",\n    sessionId                : 'YZdrO4azk',\n    serviceMsgUrl            : 'http://localhost:61180/messaging',\n    transportWorkerUrl       : 'http://localhost:61180/transport-worker.js',\n    isFirstPageLoad          : false,\n    crossDomainProxyPort     : '61181',\n    referer                  : \"\",\n    forceProxySrcForImage    : false,\n    iframeTaskScriptTemplate : {{{iframeTaskScriptTemplate}}},\n    allowMultipleWindows     : true,\n    isRecordMode             : false,\n    windowId                 : '',\n    nativeAutomation         : true,\n    disableCrossDomain       : false\n}, window);\n\n(function () {\n    (function () {\n    var IframeDriver = window['%testCafeIframeDriver%'];\n    var driver       = new IframeDriver(\"YZdrO4azk\", {\n        selectorTimeout:  10000,\n        pageLoadTimeout:  3000,\n        dialogHandler:    null,\n        retryTestPages:   false,\n        speed:            1,\n        nativeAutomation: true,\n    });\n\n    driver.start();\n\n    Object.defineProperty(window, '%testCafeDriverInstance%', {\n        configurable: true,\n        value:        driver\n    });\n})();\n\n})();\n",
    allowMultipleWindows     : true,
    isRecordMode             : false,
    windowId                 : 'tTLZga3YQ',
    nativeAutomation         : true,
    disableCrossDomain       : false
}, window);

(function () {
    (function () {
    if (window !== window.top)
        return;

    var origin = location.origin;

    // NOTE: location.origin doesn't exist in IE11 on Windows 10.10240 LTSB
    if (!origin)
        origin = location.protocol + '//' + location.hostname + (location.port ? ':' + location.port : '');

    var nativeAutomation = true;

    if (nativeAutomation)
        origin = "http://localhost:61180";

    var testRunId                                       = "YZdrO4azk";
    var browserId                                       = "C_uVTjt";
    var activeWindowId                                  = "tTLZga3YQ";
    var selectorTimeout                                 = 10000;
    var pageLoadTimeout                                 = 3000;
    var childWindowReadyTimeout                         = 30000;
    var retryTestPages                                  = false;
    var speed                                           = 1;
    var browserHeartbeatUrl                             = origin + "/browser/heartbeat/C_uVTjt";
    var browserStatusUrl                                = origin + "/browser/status/C_uVTjt";
    var browserStatusDoneUrl                            = origin + "/browser/status-done/C_uVTjt";
    var browserIdleUrl                                  = origin + "/browser/idle/C_uVTjt";
    var browserOpenFileProtocolUrl                      = origin + "/browser/open-file-protocol/C_uVTjt";
    var browserActiveWindowIdUrl                        = origin + "/browser/active-window-id/C_uVTjt";
    var browserCloseWindowUrl                           = origin + "/browser/close-window/C_uVTjt";
    var browserDispatchNativeAutomationEventUrl         = origin + "/browser/dispatch-native-automation-event/C_uVTjt";
    var browserDispatchNativeAutomationEventSequenceUrl = origin + "/browser/dispatch-native-automation-event-sequence/C_uVTjt";
    var browserParseSelectorUrl                         = origin + "/parse-selector/C_uVTjt";
    var skipJsErrors                                    = false;
    var dialogHandler                                   = null;
    var userAgent                                       = "Chrome 114.0.0.0 / Ventura 13";
    var fixtureName                                     = "test";
    var testName                                        = "tryout";
    var canUseDefaultWindowActions                      = true;

    var ClientDriver = window['%testCafeDriver%'];
    var driver       = new ClientDriver(testRunId,
        {
            heartbeat:                      browserHeartbeatUrl,
            status:                         browserStatusUrl,
            statusDone:                     browserStatusDoneUrl,
            idle:                           browserIdleUrl,
            activeWindowId:                 browserActiveWindowIdUrl,
            closeWindow:                    browserCloseWindowUrl,
            openFileProtocolUrl:            browserOpenFileProtocolUrl,
            dispatchNativeAutomationEvent:  browserDispatchNativeAutomationEventUrl,
            dispatchNativeAutomationEventSequence: browserDispatchNativeAutomationEventSequenceUrl,
            parseSelector:                  browserParseSelectorUrl,
        },
        {
            userAgent:      userAgent,
            fixtureName:    fixtureName,
            testName:       testName,
            activeWindowId: activeWindowId,
        },
        {
            selectorTimeout:            selectorTimeout,
            pageLoadTimeout:            pageLoadTimeout,
            childWindowReadyTimeout:    childWindowReadyTimeout,
            skipJsErrors:               skipJsErrors,
            dialogHandler:              dialogHandler,
            retryTestPages:             retryTestPages,
            speed:                      speed,
            canUseDefaultWindowActions: canUseDefaultWindowActions,
            nativeAutomation:                  nativeAutomation,
        }
    );

    Object.defineProperty(window, '%testCafeDriverInstance%', {
        configurable: true,
        value:        driver
    });

    driver.start();
})();

})();
</script></head><body><script class="self-removing-script-hammerhead-shadow-ui">(function () {var currentScript = document.currentScript;if (!currentScript) {var scripts       = document.scripts;var scriptsLength = scripts.length;currentScript = scripts[scriptsLength - 1];}currentScript.parentNode.removeChild(currentScript);if (window["%hammerhead%"])window["%hammerhead%"].sandbox.node.raiseBodyCreatedEvent();})();</script>{}</body></html>
```

### What is the Expected behavior?

When I mock a javascript file, the content should stay in javascript and not add html tags to it.

### What is your public website URL? (or attach your complete example)

url used in the example below: https://www.dynatrace.com/

### What is your TestCafe test code?

I created a short example to reproduce:

``` typescript
import {RequestMock} from "testcafe";

export const jsFileMock = RequestMock()
    .onRequestTo(/.*ruxitagentjs_.*\.js/g)
    .respond({}, 200);

fixture('test').page('https://www.dynatrace.com/').requestHooks(jsFileMock);

test('tryout', async (tc) => tc.wait(5000));
```

### Your complete configuration file

no config file used

### Your complete test report

![image](https://github.com/DevExpress/testcafe/assets/43231671/f2187919-61e0-4b74-ac95-c7aed6a83674)

### Steps to Reproduce

1. copy the code i put into the "What is your TestCafe test code?" section
2. run `testcafe chrome path/to/file --live`
3. open dev tools & check console
4. now you should see a `Uncaught SyntaxError: Unexpected token '<'` appearing

### TestCafe version

3.0.0

### Node.js version

16.18.1

### Command-line arguments

testcafe chrome path/to/file --live`

**Comment by miherlosev** [comment]
Hi @Thomas-Haider,

Thank you for the shared information. I've reproduced the issue.

**Comment by miherlosev** [comment]
Fixed in https://github.com/DevExpress/testcafe/pull/7827

**Comment by Thomas-Haider** [comment]
Thanks!

**Comment by Thomas-Haider** [comment]
@miherlosev is there already a timeline when this will be available?

**Comment by Aleksey28** [comment]
Hi @Thomas-Haider,
It was released in TestCafe v3.0.1.
